bftengine/src/bftengine/ControlStateManager.cpp

```diff
@@ -56,13 +56,18 @@ void ControlStateManager::addOnRestartProofCallBack(std::function<void()> cb,
 void ControlStateManager::onRestartProof(const SeqNum& seq_num, uint8_t reason) {
   // If operator sends add-remove request with bft option then
   // It can happen that some replicas receives a restart proof and yet to reach
-  // stable checkpoint. We should not rstart replica in that case since
+  // stable checkpoint. We should not restart replica in that case since
   // configuration update happens on stable checkpoint.
 
   if ((restartBftEnabled_ && IControlHandler::instance()->isOnStableCheckpoint()) ||
       IControlHandler::instance()->isOnNOutOfNCheckpoint()) {
     auto seq_num_to_stop_at = getCheckpointToStopAt();
     if (seq_num_to_stop_at.has_value() && seq_num) {
+      // A nasty hack to allow state transfer replicas to get the update in high probability - we are going to sleep for
+      // 10 seconds, assuming that it is enough for state transfer replicas to get the state change and perform the
+      // update
+      if (reason == static_cast<uint8_t>(ReplicaRestartReadyMsg::Reason::Scale))
```

Comment:
not sure i understood. why we need to sleep in case of superStableCheckpoint?

Reply:
We want to let the state transferred replica a chance to poll the latest update from the blockchain. 
Because the replicas may restart themselves (and after the restart, the state transferred replica may not be able to connect to other replicas) we wait for a while (10 seconds, where the CRE timeout is 1 second) before shutting down the replicas

Reply:
right, but if IControlHandler::instance()->isOnNOutOfNCheckpoint() == True that means, all replicas are at same checkpoint and there is no lagging replica. right?
But yes, we need it for RO replica.

Reply:
exactly